fvm/evm/handler/precompiles.go

```diff
@@ -63,21 +63,21 @@ func randomSourceProvider(contractAddress flow.Address, backend types.Backend) f
 			if types.IsAFatalError(err) || types.IsABackendError(err) {
 				panic(err)
 			}
-			return 0, err
+			return nil, err
 		}
 
 		data, ok := value.(cadence.Struct)
 		if !ok {
-			return 0, fmt.Errorf("invalid output data received from getRandomSource")
+			return nil, fmt.Errorf("invalid output data received from getRandomSource")
 		}
 
 		cadenceArray := cadence.SearchFieldByName(data, RandomSourceTypeValueFieldName).(cadence.Array)
-		source := make([]byte, 8)
+		source := make([]byte, 32)
```

Comment:
we could use the constant from [here](https://github.com/onflow/flow-go/blob/master/fvm/environment/history_random_source_provider.go#L88) to make sure the EVM source does't get trimmed in the future. 

I also suggested to update that constant to 16 to reduce the Cadence smart contract data size. Using the constant makes sure the change propagates to EVM smoothly.

Reply:
Here is my suggestion in a PR https://github.com/onflow/flow-go/pull/6170

Reply:
good idea @tarakby, fixed

Reply:
I noticed the [test](https://github.com/onflow/flow-go/blob/tarak/random-history-suggestions/fvm/environment/random_generator_test.go#L90-L96) wasn't added, but you're right that it's not related to this PR and the EVM fix. It's about the FVM side and I should have added it before. Once you merge your changes, I'll open a new PR against master and ping you for review 🙏🏼